File: tarea3/cs/metodosArreglo.cs
using System;

class metodosArreglo {
    static void Main(string[] args) {
        int[] arreglo = {0,1,2,3,4,5};

        if(args.Length==0){
            Console.WriteLine(string.Join(" ", arreglo));
            return;
        }

        if(args[0]=="insertar" && args.Length>=3){
            int i = int.Parse(args[2]);
            if(i>=0 && i<=arreglo.Length){
                int[] nuevo = new int[arreglo.Length+1];
                for(int j=0,k=0;j<nuevo.Length;j++) nuevo[j] = (j==i)? int.Parse(args[1]) : arreglo[k++];
                arreglo = nuevo;
            }
        } else if(args[0]=="extraer" && args.Length>=2){
            int i = int.Parse(args[1]);
            if(i>=0 && i<arreglo.Length){
                int[] nuevo = new int[arreglo.Length-1];
                for(int j=0,k=0;j<arreglo.Length;j++) if(j!=i) nuevo[k++] = arreglo[j];
                arreglo = nuevo;
            }
        }

        Console.WriteLine(string.Join(" ", arreglo));
    }
}

File: tarea3/metodosArreglo.java
import java.util.Arrays;

public class metodosArreglo {
    public static void main(String[] args) {
        int[] arreglo = {0, 1, 2, 3, 4, 5};

        if(args.length == 0) {
            System.out.println(Arrays.toString(arreglo));
            return;
        }
        switch (args[0]) {
            case "insertar": {
                if(args.length < 3){
                    System.out.println("Uso: insertar <valor> <indice>");
                    break;
                }
                int valor = Integer.parseInt(args[1]);
                int indice = Integer.parseInt(args[2]);
                if(indice < 0 || indice > arreglo.length){
                    System.out.println("Indice fuera de rango.");
                    break;
                }
                int[] nuevo = new int[arreglo.length + 1];
                for(int i = 0, j = 0; i < nuevo.length; i++){
                    if(i == indice){
                        nuevo[i] = valor;
                    } else {
                        nuevo[i] = arreglo[j++];
                    }
                }
                arreglo = nuevo;
                System.out.println("El nuevo arreglo es " + Arrays.toString(arreglo));
                break;
            }
            case "extraer": {
                if(args.length < 2){
                    System.out.println("Uso: extraer <indice>");
                    break;
                }
                int indice = Integer.parseInt(args[1]);
                if(indice < 0 || indice >= arreglo.length){
                    System.out.println("Indice fuera de rango.");
                    break;
                }
                int[] nuevo = new int[arreglo.length - 1];
                for(int i = 0, j = 0; i < arreglo.length; i++){
                    if(i != indice){
                        nuevo[j++] = arreglo[i];
                    }
                }
                arreglo = nuevo;
                System.out.println("El nuevo arreglo es " + Arrays.toString(arreglo));
                break;
            }
        }
    }
}
